src/community/oseo/oseo-rest/src/main/java/org/geoserver/opensearch/rest/CollectionLayerManager.java

```diff
@@ -310,6 +329,32 @@ private void configureSeparateBandsMosaic(
         createStyle(layerConfiguration, layerInfo, mosaicStoreInfo);
     }
 
+    private ImageReaderSpi getSpi(Object source) throws IOException {
+        if (source instanceof File) {
+            try (FileImageInputStream fis = new FileImageInputStream((File) source)) {
+                ImageReader imageReader = ImageIOExt.getImageioReader(fis);
+                if (imageReader != null) {
+                    return imageReader.getOriginatingProvider();
+                }
+            }
+        } else if (source instanceof SourceSPIProvider) {
+            return ((SourceSPIProvider) source).getReaderSpi();
+        }
+        return null;
+    }
+
+    private CogConfiguration getCogConfiguration(CollectionLayer layerConfiguration) {
+        CogConfiguration cogConfiguration = new CogConfiguration();
+        if (layerConfiguration != null) {
+            String rangeReader = layerConfiguration.getCogRangeReader();
```

Comment:
Wondering if there is any sort of validation about the range reader chosen... like, what if there is a typo for example?
It would be nice to get an understandable error message.

Reply:
Added some more code attempting a classforname